thrust/thrust/detail/config/cpp_compatibility.h

```diff
@@ -26,56 +26,9 @@
 #  pragma system_header
 #endif // no system header
 
-#include <thrust/detail/config/cpp_dialect.h> // IWYU pragma: export
-
-#include <cuda/std/cstddef>
-
+// deprecated [Since 2.8.0]
 #define THRUST_NODISCARD _CCCL_NODISCARD
-
-// FIXME: Combine THRUST_INLINE_CONSTANT and
-// THRUST_INLINE_INTEGRAL_MEMBER_CONSTANT into one macro when NVCC properly
-// supports `constexpr` globals in host and device code.
-#if defined(__CUDA_ARCH__) || defined(_NVHPC_CUDA)
-// FIXME: Add this when NVCC supports inline variables.
-// #  if   _CCCL_STD_VER >= 2017
-// #    define THRUST_INLINE_CONSTANT                 inline constexpr
-// #    define THRUST_INLINE_INTEGRAL_MEMBER_CONSTANT inline constexpr
-#  define THRUST_INLINE_CONSTANT                 static const _CCCL_DEVICE
-#  define THRUST_INLINE_INTEGRAL_MEMBER_CONSTANT static constexpr
-
-#else
-// FIXME: Add this when NVCC supports inline variables.
-// #  if   _CCCL_STD_VER >= 2017
-// #    define THRUST_INLINE_CONSTANT                 inline constexpr
-// #    define THRUST_INLINE_INTEGRAL_MEMBER_CONSTANT inline constexpr
-#  define THRUST_INLINE_CONSTANT                 static constexpr
-#  define THRUST_INLINE_INTEGRAL_MEMBER_CONSTANT static constexpr
-
-#endif
-
-// These definitions were intended for internal use only and are now obsolete.
-// If you relied on them, consider porting your code to use the functionality
-// in libcu++'s <nv/target> header.
-// For a temporary workaround, define THRUST_PROVIDE_LEGACY_ARCH_MACROS to make
-// them available again. These should be considered deprecated and will be
-// fully removed in a future version.
-#ifdef THRUST_PROVIDE_LEGACY_ARCH_MACROS
-#  ifndef THRUST_IS_DEVICE_CODE
-#    if defined(_NVHPC_CUDA)
-#      define THRUST_IS_DEVICE_CODE      __builtin_is_device_code()
-#      define THRUST_IS_HOST_CODE        (!__builtin_is_device_code())
-#      define THRUST_INCLUDE_DEVICE_CODE 1
-#      define THRUST_INCLUDE_HOST_CODE   1
-#    elif defined(__CUDA_ARCH__)
-#      define THRUST_IS_DEVICE_CODE      1
-#      define THRUST_IS_HOST_CODE        0
-#      define THRUST_INCLUDE_DEVICE_CODE 1
-#      define THRUST_INCLUDE_HOST_CODE   0
-#    else
-#      define THRUST_IS_DEVICE_CODE      0
-#      define THRUST_IS_HOST_CODE        1
-#      define THRUST_INCLUDE_DEVICE_CODE 0
-#      define THRUST_INCLUDE_HOST_CODE   1
-#    endif
-#  endif
-#endif // THRUST_PROVIDE_LEGACY_ARCH_MACROS
+// deprecated [Since 2.8.0]
+#define THRUST_INLINE_CONSTANT _CCCL_GLOBAL_CONSTANT
+// deprecated [Since 2.8.0]
+#define THRUST_INLINE_INTEGRAL_MEMBER_CONSTANT static constexpr
```

Comment:
Should this also add _CCCL_DEPRECATED?

Reply:
So users would find out that they rely on a deprecated macro? I think it would add more confusion, since the deprecation warning would appear in the wrong place.